services/tokenizer/core/test_types.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python3
+"""
+Tests for core tokenizer types and configuration.
+
+This module tests:
+- TokenizerConfig dataclass and validation
+- Enum types (LanguageFamily, TokenType, CaseHandling, etc.)
+- Type system edge cases and defaults
+"""
+
+from typing import Optional
+
+import pytest
+
+from .types import CaseHandling, LanguageFamily, TokenizerConfig, TokenList, TokenType
+
+
+class TestTokenizerConfig:
```

Comment:
What's the benefit of having tests implemented as classes with methods? Just wondering, I believe tests have rather different styles right now (I tend to just write functions for each test).

Reply:
I used class-based tests to group related tests logically. The class structure provides a natural namespace and makes it easier to share fixtures if needed. Happy to follow whichever convention we standardize on going forward.